xmipy/xmiwrapper.py

```diff
@@ -456,9 +460,9 @@ def get_value_ptr_scalar(self, name: str) -> NDArray:
                 byref(values),
                 detail="for variable " + name,
             )
-        elif vartype.lower().startswith("float"):
+        elif var_type_lower.startswith("float"):
             arraytype = np.ctypeslib.ndpointer(
-                dtype=float, ndim=1, shape=(1,), flags="C"
+                dtype=np.float32, ndim=1, shape=(1,), flags="C"
```

Comment:
Nice catch!

Reply:
It would be nice to include a "dummy" C and/or Fortran library to add a few more tests with. However, that alone could take quite a bit of effort.